util/device_set.cpp

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provide generic device sets support for tests that require multiple
+//  devices to run
```

Comment:
It is not clear to me why you cannot just use algorithms from https://en.cppreference.com/w/cpp/algorithm/set_difference and so on?
Sure the syntax is not fancy before C++20 ranges...

Reply:
I am not sure which logic can we re-written by using _C++ Algorithms library_, but now we using `std::copy` instead of `for each sycle`. This changes can be found in this commit: 4f771ecb8a39b3f3b855bdb362985e12a6611687